final-project/manApiProxy.tex

```diff
@@ -22,4 +22,69 @@ \section{Forwarding Requests Credentials}
-retrieve the user's permissions and session state.
\ No newline at end of file
+retrieve the user's permissions and session state.
+
+\section{OpenAPI Endpoints Route Specification}
+
+The table below shows each API endpoint and the route it will lead to. Additionally, it describes the function of each endpoint.
+
+
+\begin{table}[htbp]
+  \centering
+  \caption{API Endpoints}
+  \begin{tabular}{|c|l|c|p{6.5cm}|}
+    \hline
+    \textbf{Category} & \textbf{Endpoint} & \textbf{Methods} & \textbf{Description} \\
+    \hline
+    \multirow{6}{*}{Room} & /api/room/list & GET & List open public rooms \\
+    \cline{2-4}
+    & /api/room/{RoomName}/queue & POST, DELETE & Add one or more videos to the queue \newline Remove a video from the queue \\
+    \cline{2-4}
+    & /api/room/{RoomName} & GET, PATCH, DELETE & Get room metadata \newline Change a room's settings or claim ownership of it \newline Unload the room from memory \\
+    \cline{2-4}
+    & /api/room/generate & POST & Generate a new room \\
+    \cline{2-4}
+    & /api/room/create & POST & Create a new room \\
+    \cline{2-4}
+    & /api/room/{RoomName}/vote & POST, DELETE & Add a vote for a video \newline Remove a vote for a video \\
+    \hline
+    \multirow{6}{*}{User} & /api/user & GET & Get the current user associated with the token \\
+    \cline{2-4}
+    & /api/user/auth/discord & GET & Authenticate user with Discord \\
+    \cline{2-4}
+    & /api/user/auth/discord/callback & GET & Callback for Discord authentication \\
+    \cline{2-4}
+    & /api/user/register & POST & Register a new user \\
+    \cline{2-4}
+    & /api/user/login & POST & Log in the current user \\
+    \cline{2-4}
+    & /api/user/logout & POST & Log out the current user \\
+    \hline
+    \multirow{2}{*}{Data} & /api/data/permissions & GET & Get roles and permissions data \\
+    \cline{2-4}
+    & /api/data/previewAdd & GET & Get video preview data before adding it to the queue \\
+    \hline
+    \multirow{1}{*}{Dev} & /api/dev/reset-rate-limit & POST & Reset rate limits (development mode only) \\
+    \hline
+    \multirow{1}{*}{Auth} & /api/auth/grant & GET & Get an auth token \\
+    \cline{2-4}
+    & /api/auth/discord & GET & Authenticate with Discord \\
+    \cline{2-4}
+    & /api/auth/discord/callback & GET & Callback for Discord authentication \\
+    \hline
+    \multirow{2}{*}{Status} & /api/status & GET & Get the status of the server \\
+    \cline{2-4}
+    & /api/status/metrics & GET & Get server metrics for Prometheus \\
+    \hline
+  \end{tabular}
+\end{table}
+
+The routing between categories can be determined based on the relationships between the endpoints and the logical flow of the application. For example, endpoints within the "Room" category may have routes connecting to each other within the category, while endpoints in different categories may have separate routes. 
+
+A flowchart is shown below that demonstartes the potential path when using OpenTogetherTube.
```

Comment:
_You_ need to determine and define how these endpoints should be categorized and how they should be routed.

Basically,
1. Any `/api/room/:roomName` endpoint needs to be routed to a specific monolith
2. `/api/room/generate`, and `/api/room/create` need to be routed to the monolith with the least rooms
3. `/api/room/list` is a special case, and needs to list rooms from _all_ monoliths
4. `/api/status` and `/api/status/metrics` should _not_ be forwarded because the responses are Monolith specific, and only used for monitoring
5. All other endpoints are stateless, and can be routed to any Monolith